totaltext_converter.py 运行报错

你好，非常感谢团队的工作。有个问题
在提供的total_text下载链接中下载相应的图片和标注文件，但在运行过程中出现问题，

det：
Traceback (most recent call last):
  File "totaltext_converter.py", line 411, in <module>
    main()
  File "totaltext_converter.py", line 406, in main
    image_infos = collect_annotations(files, nproc=args.nproc)
  File "totaltext_converter.py", line 68, in collect_annotations
    images = mmcv.track_progress(load_img_info, files)
  File "/home/zhaoyinfa/.conda/envs/py37/lib/python3.7/site-packages/mmcv/utils/progressbar.py", line 92, in track_progress
    results.append(func(task, **kwargs))
  File "totaltext_converter.py", line 349, in load_img_info
    assert img.shape[0:2] == img_color.shape[0:2]
AssertionError

rec:
Converting training into instances_training.json
Loaded 1255 images from /home/data/datasets/OCR/rec/Totaltext/imgs/training
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 1255/1255, 64.3 task/s, elapsed: 20s, ETA:     0s
It takes 55.86578726768494s to convert totaltext annotation
Converting test into instances_test.json
Loaded 300 images from /home/data/datasets/OCR/rec/Totaltext/imgs/test
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 300/300, 54.6 task/s, elapsed: 5s, ETA:     0s
It takes 15.043917655944824s to convert totaltext annotation
Traceback (most recent call last):
  File "totaltext_converter.py", line 380, in <module>
    main()
  File "totaltext_converter.py", line 376, in main
    generate_ann(root_path, split, image_infos)
  File "totaltext_converter.py", line 300, in generate_ann
    mmcv.imwrite(dst_img, dst_img_path)
  File "/home/zhaoyinfa/.conda/envs/py37/lib/python3.7/site-packages/mmcv/image/io.py", line 258, in imwrite
    return cv2.imwrite(file_path, img, params)
cv2.error: OpenCV(4.5.3) /tmp/pip-req-build-l1r0y34w/opencv/modules/imgcodecs/src/loadsave.cpp:803: error: (-215:Assertion failed) !_img.empty() in function 'imwrite'

--- a/tools/data/textrecog/totaltext_converter.py
+++ b/tools/data/textrecog/totaltext_converter.py
@@ -294,6 +294,9 @@ def generate_ann(root_path, split, image_infos):
         for anno in image_info['anno_info']:
             word = anno['word']
             dst_img = crop_img(image, anno['bbox'])
+            # skip invalid annotations
+            if dst_img.shape[0] == 0 or dst_img.shape[1] == 0:
+                continue
             dst_img_name = f'{src_img_root}_{index}.png'
             index += 1
             dst_img_path = osp.join(dst_image_root, dst_img_name)

--- a/tools/data/textrecog/totaltext_converter.py
+++ b/tools/data/textrecog/totaltext_converter.py
@@ -294,9 +294,11 @@ def generate_ann(root_path, split, image_infos):
         for anno in image_info['anno_info']:
             word = anno['word']
             dst_img = crop_img(image, anno['bbox'])
-            # skip invalid annotations
-            if dst_img.shape[0] == 0 or dst_img.shape[1] == 0:
+
+            # Skip invalid annotations
+            if min(dst_img.shape) == 0:
                 continue
+
             dst_img_name = f'{src_img_root}_{index}.png'
             index += 1
             dst_img_path = osp.join(dst_image_root, dst_img_name)

